src/status_im/contexts/wallet/events.cljs

```diff
@@ -65,16 +65,27 @@
    {:db (assoc-in db [:wallet :current-viewing-account-address] address)}))
 
 (rf/reg-event-fx :wallet/clean-current-viewing-account
- (fn [{:keys [db]}]
-   (let [just-completed-transaction? (get-in db [:wallet :ui :send :just-completed-transaction?])]
-     (when-not just-completed-transaction?
-       {:db (update db :wallet dissoc :current-viewing-account-address)}))))
+ (fn [{:keys [db]} [ignore-just-completed-transaction?]]
+   (let [{:keys [entry-point just-completed-transaction?]} (-> db :wallet :ui :send)
+         entry-point-wallet-home?                          (= entry-point :wallet-home)]
+     {:db (cond-> db
+            (and (not entry-point)
+                 (not ignore-just-completed-transaction?)
+                 (not just-completed-transaction?))
+            (update :wallet dissoc :current-viewing-account-address)
+
+            entry-point-wallet-home?
+            (update-in [:wallet :ui :send] dissoc :entry-point)
+
+            (and entry-point-wallet-home?
+                 (not just-completed-transaction?))
+            (update :wallet dissoc :current-viewing-account-address))})))
```

Comment:
This event was complex and now it's more complex :cry: 

This event will remove entries from re-frame when we navigate back but it depends on trhe `entry-point`.

Currently, `:current-viewing-account` is being used as the sender address, this shouldn't have been made like this, because when we remove `current-viewing-account` from the DB, it also affects the UI.

A larger refactor will be needed if we want to simplify the logic, but it's out of scope of this PR.

CC: @status-im/wallet-mobile-devs